pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -375,6 +375,11 @@ func makeHookPod(hook *deployapi.LifecycleHook, rc *kapi.ReplicationController,
 
 	gracePeriod := int64(10)
 
+	podSecurityContext, err := kapi.Scheme.DeepCopy(rc.Spec.Template.Spec.SecurityContext)
```

Comment:
you have to tell the hook pod the container name so it pulls the image.  That would be unique